nifi-nar-bundles/nifi-iaa-providers-bundle/nifi-file-identity-provider/src/main/java/org/apache/nifi/authentication/file/FileIdentityProvider.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    @Override
+    public final void onConfigured(final LoginIdentityProviderConfigurationContext configurationContext) throws ProviderCreationException {
+        final Map<String, String> configProperties = configurationContext.getProperties();
+        for (String propertyKey : configProperties.keySet()) {
+            String propValue = configProperties.get(propertyKey);
+            logger.debug("found property '{}': '{}'", propertyKey, propValue);
+        }
+
+        credentialsFilePath = configProperties.get(PROPERTY_CREDENTIALS_FILE);
+        if (credentialsFilePath == null) {
+            final String message = String.format("Identity Provider '%s' requires a credentials file path in property '%s'",
+                    identifier, PROPERTY_CREDENTIALS_FILE);
+            throw new ProviderCreationException(message);
+        }
+
+        final String rawExpirationPeriod = configProperties.get(PROPERTY_EXPIRATION_PERIOD);
+        if (rawExpirationPeriod == null || rawExpirationPeriod.isEmpty()) {
+            final String message = String.format("Identity Provider '%s' requires a credential expiration in property '%s'",
+                    identifier, PROPERTY_EXPIRATION_PERIOD);
+            throw new ProviderCreationException(message);
+        } else {
+            try {
+                expirationPeriodMilliseconds = FormatUtils.getTimeDuration(rawExpirationPeriod, TimeUnit.MILLISECONDS);
+            } catch (IllegalArgumentException iae) {
+                final String message = String.format("Identity Provider '%s' property '%s' value of '%s', is not a valid time period",
+                        identifier, PROPERTY_EXPIRATION_PERIOD, rawExpirationPeriod);
+                throw new ProviderCreationException(message);
+            }
+        }
+
+        logger.debug("Identity Provider '{}' configured to use file '{}' and expiration period of '{}'={} milliseconds",
+                identifier, credentialsFilePath, rawExpirationPeriod, expirationPeriodMilliseconds);
+    }
+
+    String getCredentialsFilePath() {
+        return credentialsFilePath;
+    }
+
+    long getExpirationPeriod() {
+        return expirationPeriodMilliseconds;
+    }
+
+    @Override
+    public final AuthenticationResponse authenticate(final LoginCredentials credentials) throws InvalidLoginCredentialsException, IdentityAccessException {
+        final String loginUsername = credentials.getUsername();
+        final String loginPassword = credentials.getPassword();
+        AuthenticationResponse authResponse = null;
+
+        try {
+            UserCredentialsList credentialsList = loadCredentialsList(credentialsFilePath);
+            List<UserCredentials> userCredentials = credentialsList.getUser();
+            for (UserCredentials userCreds : userCredentials) {
+                final String storedUsername = userCreds.getName();
+                if (storedUsername.equals(loginUsername)) {
```

Comment:
I will add an XSD unique constraint on the usernames, and a unit test for the validation.  It looks like XSD constraints do not (easily?) do case-insensitive unique comparisons, so it won't be foolproof, but it may help prevent some accidents.